engine/consensus/approvals/assignment_collector_tree.go

```diff
@@ -235,20 +251,20 @@ func (t *AssignmentCollectorTree) GetOrCreateCollector(result *flow.ExecutionRes
 	v, found := t.forest.GetVertex(resultID)
 	if found {
 		return &LazyInitCollector{
-			Collector:   v.(*assignmentCollectorVertex).collector,
-			Processable: v.(*assignmentCollectorVertex).processable,
-			Created:     false,
+			Collector: v.(*assignmentCollectorVertex).collector,
+			Created:   false,
 		}, nil
 	}
 
 	// An assignment collector is processable if and only if:
 	// either (i) the parent result is the latest sealed result (seal is finalized)
 	//    or (ii) the result's parent is processable
 	parent, parentFound := t.forest.GetVertex(result.PreviousResultID)
+	newStatus := CachingApprovals
 	if parentFound {
-		vertex.processable = parent.(*assignmentCollectorVertex).processable
+		newStatus = parent.(*assignmentCollectorVertex).collector.ProcessingStatus()
 	} else if executedBlock.ParentID == t.lastSealedID {
```

Comment:
I think the collector should be in state `VerifyingApprovals` if `executedBlock.ParentID == t.lastSealedID`:

```suggestion
	}
	if executedBlock.ParentID == t.lastSealedID {
```